schema_based_on_diagram.sql

```diff
@@ -0,0 +1,52 @@
+CREATE TABLE patients (
+	id INT GENERATED ALWAYS AS IDENTITY,
+	name VARCHAR(100),
+	date_of_birth DATE,
+	PRIMARY KEY(id)
+);
+
+CREATE TABLE medical_histories (
+	id INT GENERATED ALWAYS AS IDENTITY,
+	admitted_at DATE,
+	status VARCHAR(100),
+	patient_id INT,
+	PRIMARY KEY(id),
+	CONSTRAINT fk_patients FOREIGN KEY(patient_id) REFERENCES patients(id)
+);
+
+CREATE TABLE treatments (
+	id INT GENERATED ALWAYS AS IDENTITY,
+	type VARCHAR(100),
+	name VARCHAR(100),
+	PRIMARY KEY (id)
+);
+
+CREATE TABLE invoices (
+	id INT GENERATED ALWAYS AS IDENTITY,
+	total_amount DECIMAL,
+	generated_at TIMESTAMP,
+	payed_at TIMESTAMP,
+	medical_history_id INT,
+	PRIMARY KEY(id),
+	CONSTRAINT fk_med_history FOREIGN KEY(medical_history_id) REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_items (
+	id INT GENERATED ALWAYS AS IDENTITY,
+	unit_price DECIMAL,
+	quantity INT,
+	total_price DECIMAL,
+	invoice_id INT,
+	treatment_id INT,
+	PRIMARY KEY (id),
+	CONSTRAINT fk_invoice FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+	CONSTRAINT fk_treatment FOREIGN KEY (treatment_id) REFERENCES treatments(id),
+);
+
+
+
+CREATE INDEX idx_medical_histories_patient_id ON medical_histories(patient_id);
+CREATE INDEX idx_invoices_medical_history_id ON invoices(medical_history_id);
+CREATE INDEX idx_invoice_items_invoice_id ON invoice_items(invoice_id);
+CREATE INDEX idx_invoice_items_treatment_id ON invoice_items(treatment_id);
```

Comment:
Two more indexes are required.

There is a relationship between medical history and treatment which is represented by `medical_history_has_treatment`

```
CREATE INDEX ON medical_histories_has_treatments (medical_history_id);
CREATE INDEX ON medical_histories_has_treatments (treatment_id);
```

Reply:
Thanks for flagging this, @brytebee